### PR TITLE
fix for issue #679

### PR DESCRIPTION
**Pull request by Beeblerox**

### CODE REVIEW

File: flixel/util/FlxTimer.hx [hunk 1]
```diff
@@ -156,16 +156,23 @@ class FlxTimer implements IFlxDestroyable
...
-				cancel();
-			}
+		}
+	}
+	
+	/**
+	 * Called by timer manager when timer finishes loop.
+	 */
+	@:allow(flixel.util.FlxTimerManager)
+	private inline function onLoop():Void
```
Comment (Gama11) [inline flixel/util/FlxTimer.hx]: I don't think this should be inline?

**Comment by Gama11** [comment]
A unit test for this would be nice. I think we might already have one for tweens where we had the same issue?

**Comment by Beeblerox** [comment]
@Gama11 cant' find test for that issue with tweens, and have no idea how to implement it

File: flixel/util/FlxTimer.hx [hunk 2]
```diff
@@ -156,16 +156,23 @@ class FlxTimer implements IFlxDestroyable
...
-				cancel();
-			}
+		}
+	}
+	
+	/**
+	 * Called by timer manager when timer finishes loop.
+	 */
+	@:allow(flixel.util.FlxTimerManager)
+	private function onLoop():Void
```
Comment (Gama11) [inline flixel/util/FlxTimer.hx]: Should be `onLoopFinished()`, which I guess also makes the doc comment obsolete. :)

File: flixel/util/FlxTimer.hx [hunk 3]
```diff
@@ -232,11 +241,31 @@ class FlxTimerManager extends FlxBasic
 	 */
 	override public function update(elapsed:Float):Void
 	{
+		var timerLoops:Int;
```
Comment (Gama11) [inline flixel/util/FlxTimer.hx]: I prefer to declare local vars as close to where they're actually needed as possible.

**Comment by Gama11** [comment]
Closing; this fix causes `FlxTimerTest` to go into an infinite loop (which is why Travis was failing here).

**Pull request closed without merging**